core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -244,6 +244,31 @@ class SparkHadoopUtil extends Logging {
   }
 
   /**
+   * List directories/files matching the path and return the `FileStatus` results.
+   * If the pattern is not a regexp then a simple `getFileStatus(pattern)`
+   * is called to get the status of that path.
+   * If the path/pattern does not match anything in the filesystem,
+   * an empty sequence is returned.
+   * @param pattern pattern
+   * @return a possibly empty array of FileStatus entries
+   */
+  def globToFileStatus(pattern: Path): Array[FileStatus] = {
```

Comment:
Yea, but then that's wrong if for example my path actually has a ? or ^ or ] in it. It doesn't seem essential and seems even problematic to add this behavior change to an otherwise clear fix.